src/Processors/Formats/Impl/MySQLDumpRowInputFormat.cpp

```diff
@@ -0,0 +1,467 @@
+/// Like '(x, `column name`, z)'
+void tryReadColumnNames(ReadBuffer & in, std::vector<String> * column_names)
+{
+    skipWhitespaceIfAny(in);
+    /// Check that we have the list of columns.
+    if (in.eof() || *in.position() != '(')
+        return;
+
+    ++in.position();
+    skipWhitespaceIfAny(in);
+    bool first = true;
+    while (!in.eof() && *in.position() != ')')
+    {
+        if (!first)
+        {
+            assertChar(',', in);
+            skipWhitespaceIfAny(in);
+        }
+
+        first = false;
+
+        String name;
+        if (!in.eof() && *in.position() == '`')
+            readBackQuotedString(name, in);
+        else
+            readUnquotedColumnName(name, in);
+
+        skipWhitespaceIfAny(in);
+
+        if (column_names)
+            column_names->push_back(name);
+    }
+
+    assertChar(')', in);
+}
+
+static MySQLQueryType skipToInsertOrCreateQuery(String & table_name, ReadBuffer & in, bool skip_create_query = false)
+{
+    String current_table_name;
+    String keyword;
+    MySQLQueryType type = MySQLQueryType::NONE;
+    /// In MySQL dumps INSERT queries might be replaced with REPLACE queries.
+    std::unordered_set<String> keywords = {"insert", "replace"};
+    if (!skip_create_query)
+        keywords.insert("create");
+
+    do
+    {
+        if (!skipUntilRowStartedWithOneOfKeywords(keywords, in, &keyword))
+            return MySQLQueryType::NONE;
+
+        skipWhitespaceIfAny(in);
+        if (keyword != "create")
+        {
+            assertStringCaseInsensitive("into", in);
+            type = MySQLQueryType::INSERT;
+        }
+        else
+        {
+            /// Check that it's CREATE TABLE query.
+            if (!checkStringCaseInsensitive("table", in))
+                continue;
+            type = MySQLQueryType::CREATE;
+        }
+
+        current_table_name = readTableName(in);
+        if (table_name.empty())
+            table_name = current_table_name;
+    } while (current_table_name != table_name);
+
+    /// Note that now position in the buffer is located in CREATE/INSERT query after table name.
+    return type;
+}
+
+static bool skipToInsertQuery(String & table_name, ReadBuffer & in)
+{
+    return skipToInsertOrCreateQuery(table_name, in, true) != MySQLQueryType::NONE;
+}
+
+static void skipToDataInInsertQuery(ReadBuffer & in, std::vector<String> * column_names = nullptr)
+{
+    tryReadColumnNames(in, column_names);
+    skipWhitespaceIfAny(in);
+    assertStringCaseInsensitive("values", in);
+    skipWhitespaceIfAny(in);
+}
+
+static bool tryToExtractStructureFromCreateQuery(ReadBuffer & in, ContextPtr & context, NamesAndTypesList & structure)
+{
+    /// To extract structure from CREATE query we read it all into a string and parse using MySQLParser
+    /// and then extract columns names and types from MySQLParser::ASTCreateDefines
+
+    /// Now position is located in CREATE query after table name and we need to write the beginning of CREATE query.
```

Comment:
Is it because we have read it just to know table name?

Reply:
Yes, we need to know table name to compare it with name specified in settings (or with first table name in the dump)